packages/nuxt/src/pages/utils.ts

```diff
@@ -128,25 +134,74 @@ export function extractScriptContent (html: string) {
 
 const PAGE_META_RE = /(definePageMeta\([\s\S]*?\))/
 
-async function getRouteName (file: string) {
-  const script = extractScriptContent(file)
-  if (!script) { return null }
+const metaCache: Record<string, Partial<Record<keyof NuxtPage, any>>> = {}
+async function getRouteMeta (contents: string, absolutePath?: string): Promise<Partial<Record<keyof NuxtPage, any>>> {
```

Comment:
I think `absolutePath` is always passed to `getRouteName`, would using it as key for `metaCache` be more performant?

Reply:
I think the metadata might vary based on the contents of the file. (So unless we can invalidate based on a fs watcher, we probably can't key to the file name...) But we could consider another approach, like keying to a hash of the file contents.

Reply:
>I think the metadata might vary based on the contents of the file. 

Ah, of course 😅..

>But we could consider another approach, like keying to a hash of the file contents.

I'm not sure how performance compares between using large key or hashing large strings, my guess is that the current approach would do better (unless the file hash is already available someplace else). 

Maybe something in between could be to combine `absolutePath` and `script` (hashed or not, a placeholder if no script is present) to use as a key? But then `extractScriptContent` would always need to be called 😅